File: solutions/java/rotational-cipher/1/src/main/java/RotationalCipher.java
class RotationalCipher {

    private int shiftKey;

    RotationalCipher(int shiftKey) {
        this.shiftKey = shiftKey;
    }

    String rotate(String data) {
        StringBuilder sb = new StringBuilder();
        for (char c : data.toCharArray()) {
            if (Character.isLetter(c)) {
                int offset = Character.isLowerCase(c) ? 97 : 65;
                c = (char) ((((c - offset) + this.shiftKey) % 26) + offset);
            }
            sb.append(c);
        }
        return sb.toString();
    }

}

File: solutions/cpp/rotational-cipher/1/rotational_cipher.cpp
#include "rotational_cipher.h"

namespace rotational_cipher {

std::string rotate(std::string plain, int key) {
    std::string encrypted {};

    for (char ch : plain) {
     char ech {};
     if(ch >= 'a' && ch <= 'z')
         ech = (ch - 'a' + key) % 26 + 'a';
     else if (ch >= 'A' && ch <= 'Z')
         ech = (ch - 'A' + key) % 26 + 'A';
     else 
         ech = ch;

        encrypted.push_back(ech);
    }
    
    return encrypted;
}

}  // namespace rotational_cipher
